utils.py

```diff
@@ -9,6 +9,188 @@
+    """
+    months = [
+        'january',
+        'february',
+        'march',
+        'april',
+        'may',
+        'june',
+        'july',
+        'august',
+        'september',
+        'october',
+        'november',
+        'december']
+    best_score = 0
+    best_result = ''
+    for month in months:
+        if abs(len(month) - len(word)) > 1:
+            continue
+        result, score = sliding_window(word, month)
+        # print(result, score)
+        if score > best_score:
+            best_score = score
+            best_result = month
+    return (best_result, best_score)
+
+
+def make_typo_dict(word_list=[]):
+    """
+    Creates a dictionary with keys equal to word_list, with values being a 
+        list of every possible single-letter deletion of that word
+
+    Args:
+        word_list (list): a list of strings to make single deletions from. If
+            none is provided (i.e., an empty list) is uses the keys from the
+            month_dict below.
+    Returns:
+        typo_dict (dict): dictionary containing every possible single-letter
+            deletion of each of the keys in the dict    
+    """
+    month_dict = {'january': '01',
+                  'february': '02',
+                  'march': '03',
+                  'april': '04',
+                  'may': '05',
+                  'june': '06',
+                  'july': '07',
+                  'august': '08',
+                  'september': '09',
+                  'october': '10',
+                  'november': '11',
+                  'december': '12'
+                }
+    if len(word_list) == 0:
+        word_list = month_dict.keys()
+
+    # this creates a dictionary containing all possible single-letter deletions of every month
+    # it's used to correct for typos in the text of a link to the minutes
+    # ex: Novembe 17, 2010 (one of two real examples as of Aug 9, 2020)
+    typo_dict = {k: list() for k in word_list}
+    for month in typo_dict.keys():
+        for i in range(len(month)):
+            typo_dict[month].append(month[:i] + month[i+1:])
+    
+    return typo_dict
+
+
+def pick_best_month(word, typ_dict=make_typo_dict()):
+    """
+    'Wrapper' function for various spell-checking methods. Check deletions
+        first to avoid known errors with deletions and sliding_window.
+
+    Args:
+        word (str): the misspelled month string
+        typo_dict (dict): dictionary with months for keys, values are lists
+            of all single-letter deletions for that month (key)
+
+    Returns:
+        tuple(best_match, best_match_score)
+            best_match (str): methods best guess at what the correct month is
+            best_match_score (int): numeric score for how good the match is
+                higher is better, can be normalized to len(month)
+    """
+    # returns a tuple (best match, match score)
+    if match := check_dels(word, typ_dict):
```

Comment:
@robertnunn Also just a side note that the `:=` assignment expression [only works for python 3.8 or later](https://docs.python.org/3/whatsnew/3.8.html), which would mean we'd need to update the `README.md` to specifically require that version of python as a pre-requisite for running this project locally. 

It is a pretty handy operator (and gotta love the nickname "walrus operator") but would hate for someone to be banging their head trying to get the code to run because of a simple versioning syntax issue.